src/main/kotlin/ecommerce/service/PaymentService.kt

```
@@ -0,0 +1,69 @@
+package ecommerce.service
+
+import ecommerce.client.StripeClient
+import ecommerce.dto.OrderRequest
+import ecommerce.dto.PaymentResponse
+import ecommerce.enum.PaymentStatus
+import ecommerce.exception.StripePaymentException
+import ecommerce.model.Payment
+import ecommerce.repository.PaymentJpaRepository
+import ecommerce.repository.getByPaymentIntentIdOrThrow
+import jakarta.transaction.Transactional
+import org.hibernate.service.spi.ServiceException
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+
+@Transactional
+@Service
+class PaymentService(
+    private val stripeClient: StripeClient,
+    private val paymentJpaRepository: PaymentJpaRepository,
+) {
+    fun createPaymentIntent(
+        request: OrderRequest,
+        amount: BigDecimal,
+    ): PaymentResponse {
+        try {
+            val paymentResponse =
+                stripeClient.createCheckoutSession(request, amount.multiply(BigDecimal.valueOf(100)).toInt())
```

Nice use of `BigDecimal` when working with money! 🤑 
However, if we `toInt()` it, there would still be an edge case that we should consider.
How do you think about this?

---

I hadn’t thought about that, using toInt here can silently truncate decimals and even overflow for large amounts.
A safer approach would be to round explicitly and use Long instead.